js/functions.js

```diff
@@ -1,18 +1,60 @@
+//array to check the operations.
+const arrayOfOperations = [
+  "1",
+  "2",
+  "3",
+  "4",
+  "5",
+  "6",
+  "7",
+  "8",
+  "9",
+  "0",
+  ".",
+  "+",
+  "-",
+  "/",
+  "*",
+  "**",
+  "!",
+  "**2",
+  "**3",
+  "%",
+  "rt",
+  "(",
+  ")",
+  "e",
+  "π",
+  "log1",
+  "log2",
+  "ln",
+  "e**",
+];
+
+//object to track the unit that changes on click
+let unitOfAngle = {
+  degree: true,
+  radian: false,
+  grad: false,
+};
```

Comment:
Is it possible to simplify this logic ?
I think it is very much possible to introduce impossible states here.
For example,
```js
unitOfAngle = { degree: true, radian: true, grad: false }
```

In this we need to make sure only one value remains true at a time, If more than one values become true at a time, it can introduce bugs.

Reply:
will change it in the ts project with the single value of string and handle it in the if else part.